### PR TITLE
Todo List Project

### PR DESCRIPTION
**Pull request by AymenBida**
### Todo List Project

In this project we did: 

- create Todo and Project classes for holding tasks and projects objects
- create forms to add tasks and projects
- create tables to show tasks and projects
- use local storage to store the data

### CODE REVIEW

File: src/todo.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+    return taskInputs;
+  }
+
+  addEditEvent(button) {
+    button.addEventListener('click', () => {
+      this.populateTaskForm();
+      const editTaskForm = document.getElementById('editTaskForm');
+      utils.addHiddenInput(editTaskForm, this);
+      utils.toggleShowElement(editTaskForm);
```
Comment (whiz25) [inline src/todo.js]: - DOM manipulation logic should be separated from pure JS. Move any DOM logic to different file.

File: src/project.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    const tr = document.createElement('tr');
+    tr.dataset.projectId = this.index;
+    const td = document.createElement('td');
+    const tdExtra = document.createElement('td');
+    td.textContent = this.title;
+    const button = this.showTasksbutton();
+    tr.appendChild(td);
+    tr.appendChild(button);
+    tr.appendChild(tdExtra);
+    return tr;
```
Comment (whiz25) [inline src/project.js]: - Move this code to a DOM manipulation module.

File: src/utility.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
+import Project from './project';
+
+const projectsList = document.getElementById('project');
+const editProjectsList = document.getElementById('editProject');
+
+function toggleShowElement(element) {
```
Comment (whiz25) [inline src/utility.js]: - [x] The `function` syntax should be replaced with ES6 syntax.
```javascript
const toggleShowElement = () => {}
```
- [x] Do the same for all functions

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 
Nice work on this project. I was able to create a new project and add tasks to a given project. Well done!
Let's make some changes before merging.
- [x] Use ES6 syntax. Check my comments.
- [x] Separate DOM logic from business logic. Check comments.
- [x] Fix the edit feature. Clicking the `Edit` button for an individual task doesn't render the task for editing. Please fix that.


### Martin Afani
#### Slack: whiz25

**Review by NtwaliHeritier — APPROVED** [review]
### **STATUS: APPROVED✔✔✔**

Hello team,
This is a great project.

Great work making the required changes.
You can merge👍👍👍

Happy coding 💻💻💻

**Pull request closed without merging**